src/ray/common/buffer.h

```diff
@@ -51,36 +45,20 @@ class LocalMemoryBuffer : public Buffer {
   /// \param size The size of the passed in buffer.
   /// \param copy_data If true, data will be copied and owned by this buffer,
   ///                  otherwise the buffer only points to the given address.
-  LocalMemoryBuffer(uint8_t *data, size_t size, bool copy_data = false)
-      : has_data_copy_(copy_data) {
-    if (copy_data) {
-      RAY_CHECK(data != nullptr);
-      buffer_.resize(size);
-      std::copy(data, data + size, buffer_.begin());
-      data_ = buffer_.data();
-      size_ = buffer_.size();
-    } else {
-      data_ = data;
-      size_ = size;
-    }
-  }
+  LocalMemoryBuffer(uint8_t *data, size_t size, bool copy_data = false);
 
   /// Construct a LocalMemoryBuffer of all zeros of the given size.
-  LocalMemoryBuffer(size_t size) : has_data_copy_(true) {
-    buffer_.resize(size, 0);
-    data_ = buffer_.data();
-    size_ = buffer_.size();
-  }
+  LocalMemoryBuffer(size_t size);
 
-  uint8_t *Data() const override { return data_; }
+  inline uint8_t *Data() const override { return data_; }
```

Comment:
Let's remove the "inline" keyword from the methods that are already inlined (it only adds clutter).